### DISCUSSION TITLE
Example how to use this library to modify a state

### DISCUSSION THREAD

**Discussion by derjoerg** [kingsleyadam/local-abbfreeathome#33] [Q&A] [answered]
Hi,

would it be possible to add an additional example in the README how to modify a state of a device with this library (e.g. turn on/off a switch)? Currently there are only examples on how to get information.

**Comment by kingsleyadam** [comment]
Yep, I can describe the overall structure of the library and how the classes should interact with the api.

**Comment by kingsleyadam** [comment]
Here's a PR that should hopefully help, let me know if it needs more clarity. https://github.com/kingsleyadam/local-abbfreeathome/pull/35

**Reply by derjoerg** [reply]
Cool,

this makes a lot of stuff more clearer. Just one thing I'm still struggeling with is all this async stuff in python.

To turn on a switch through your library I modified your websocket-example as follow:
```
    # Set the callback function on each switch.
    for _switch in _free_at_home.get_device_by_class(device_class=SwitchActuator):
        _switch.register_callback(my_very_own_callback)
        if _switch.channel_name == "Treppe KG ":
            asyncio.run(_switch.turn_on())
```
Would this be a correct (rather simple) use-case of your library?

**Reply by kingsleyadam** [reply, marked as answer]
Yep, you wouldn't need to register a call-back if you're just turning on a switch, so even simplier.

```python
for _switch in _free_at_home.get_device_by_class(device_class=SwitchActuator):
    if _switch.channel_name == "Treppe KG ":
        asyncio.run(_switch.turn_on())
```

This is of course a bit flaky because of the use of channel_name which could change, so I'd probably use device id and channel id. But this is a working example of how to call it in a one-off script.
